src/functions/schema-path.ts

```diff
@@ -37,5 +37,11 @@ export const schemaPath: IFunction<ISchemaPathOptions> = (targetVal, opts, paths
     console.error(error);
   }
 
-  return schema(relevantObject, { schema: schemaObject }, paths, otherValues);
+  return schema(relevantObject, { schema: withoutRequiredIfNotArray(schemaObject) }, paths, otherValues);
 };
+
+function withoutRequiredIfNotArray(schemaObject: any) {
```

Comment:
Is this required or is this an older change that is left in here? 

This should not need to be done. If `required: true` is there it means the rule is operating on a parameter object, which with the jsonpath rules you've set up, it shouldnt ever be. 

The only way a schema object will have a required: true in it, is because the user has invalid API specifications, and that'll get caught earlier.

Reply:
hi @phil :)
so `oas3-schema` rule lints a spec with:
```yaml
     parameters:
        - name: petId
          in: path
          required: true
          description: The id of the pet to retrieve
          schema:
            type: string
            example: "2323"
          example: "2323"
```
as OK. So, I'm assuming this is valid OAS3. Maybe it isn't?
Now:
1. `valid-openapi-example` will match `schema` and `schema.example`
2. `valid-example` will match `parameters[0]` and `parameters[0].example`

The problem is with `2`, since there is `required: true` on the object, that is later passed to https://github.com/stoplightio/spectral/blob/6590414fc69777e58c084463c83c11092aaccab6/src/functions/schema.ts#L61, and that's why I'm using `withoutRequiredIfNotArray`. Maybe `oas3-schema` should lint this case as OK ?

Reply:
this is entirely valid OAS3 yeah.

1. valid-schema-example should match example against the `schema` object
1. valid-openapi-example should match example against the current (`$.`) object

Reply:
@phil, done, `withoutRequiredIfNotArray` is removed now